routing/manager/manager.go

```diff
@@ -0,0 +1,601 @@
+		}
+		if neighbor.State == StateACK && !neighbor.DiffBuff.IsEmpty() {
+			r.sendRoutingMessage(&lnwire.NeighborUpdMessage{DiffBuff: neighbor.DiffBuff.Copy()}, neighbor.Id)
+			neighbor.DiffBuff.Clear()
+			neighbor.State = StateWAIT
+		}
+	}
+}
+
+func (r *RoutingManager) AddChannelAsync(Id1, Id2 graph.ID, TxID graph.EdgeID, info *graph.ChannelInfo) <-chan error {
+	msg := &addChannelCmdMessage{
+		Id1:  Id1,
+		Id2:  Id2,
+		TxID: TxID,
+		Info: info,
+		err:  make(chan error, 1),
+	}
+	r.ChIn <- msg
+	return msg.err
+}
+
+func (r *RoutingManager) AddChannel(Id1, Id2 graph.ID, TxID graph.EdgeID, info *graph.ChannelInfo) error {
+	return <-r.AddChannelAsync(Id1, Id2, TxID, info)
+}
+
+func (r *RoutingManager) HasChannelAsync(Id1, Id2 graph.ID, TxID graph.EdgeID) <-chan bool {
+	msg := &hasChannelCmdMessage{
+		Id1:  Id1,
+		Id2:  Id2,
+		TxID: TxID,
+		rez:  make(chan bool, 1),
+		err:  make(chan error, 1),
+	}
+	r.ChIn <- msg
+	return msg.rez
+}
+
+func (r *RoutingManager) HasChannel(Id1, Id2 graph.ID, TxID graph.EdgeID) bool {
+	return <-r.HasChannelAsync(Id1, Id2, TxID)
+}
+
+func (r *RoutingManager) RemoveChannelAsync(Id1, Id2 graph.ID, TxID graph.EdgeID) <-chan error {
+	msg := &removeChannelCmdMessage{
+		Id1:  Id1,
+		Id2:  Id2,
+		TxID: TxID,
+		err:  make(chan error, 1),
+	}
+	r.ChIn <- msg
+	return msg.err
+}
+
+func (r *RoutingManager) RemoveChannel(Id1, Id2 graph.ID, TxID graph.EdgeID) error {
+	return <-r.RemoveChannelAsync(Id1, Id2, TxID)
+}
+
+func (r *RoutingManager) OpenChannelAsync(Id graph.ID, TxID graph.EdgeID, info *graph.ChannelInfo) <-chan error {
+	msg := &openChannelCmdMessage{
+		Id:   Id,
+		TxID: TxID,
+		info: info,
+		err:  make(chan error, 1),
+	}
+	r.ChIn <- msg
+	return msg.err
+}
+
+func (r *RoutingManager) OpenChannel(Id graph.ID, TxID graph.EdgeID, info *graph.ChannelInfo) error {
+	return <-r.OpenChannelAsync(Id, TxID, info)
+}
+
+func (r *RoutingManager) RequestRoutingTableAsync(Id graph.ID, Timeout time.Duration) (<-chan *rt.RoutingTable, <-chan error) {
+	msg := &requestRoutingTableCmdMessage{
+		Id:      Id,
+		timeout: Timeout,
+		rez:     make(chan *rt.RoutingTable, 1),
+		err:     make(chan error, 1),
+	}
+	r.ChIn <- msg
+	return msg.rez, msg.err
+}
+
+func (r *RoutingManager) RequestRoutingTable(Id graph.ID, Timeout time.Duration) (*rt.RoutingTable, error) {
+	rez, err := r.RequestRoutingTableAsync(Id, Timeout)
+	return <-rez, <-err
+}
+
+func (r *RoutingManager) FindPathAsync(destId graph.ID, timeout time.Duration) (<-chan []graph.ID, <-chan error) {
+	msg := &findPathCmdMessage{
+		Id:      destId,
+		timeout: timeout,
+		rez:     make(chan []graph.ID, 1),
+		err:     make(chan error, 1),
+	}
+	r.ChIn <- msg
+	return msg.rez, msg.err
+}
+
+func (r *RoutingManager) FindPath(destId graph.ID, timeout time.Duration) ([]graph.ID, error) {
+	rez, err := r.FindPathAsync(destId, timeout)
+	return <-rez, <-err
+}
+
+func (r *RoutingManager) handleFindPath(msg *findPathCmdMessage) {
+	rtTemp := r.RT.Copy()
```

Comment:
rtTemp is modified when responses from other nodes arrive. mutex will not solve this.
Copy on write can be used here but it will make code slightly more complex.